packages/account/src/providers/provider.ts

```diff
@@ -521,10 +553,93 @@ export default class Provider {
         fullRequest = await options.requestMiddleware(fullRequest);
       }
 
-      return options.fetch ? options.fetch(url, fullRequest, options) : fetch(url, fullRequest);
+      if (Provider.ENSURE_RPC_CONSISTENCY) {
+        Provider.applyBlockHeight(fullRequest, url);
+      }
+
+      return Provider.fetchAndProcessBlockHeight(url, fullRequest, options);
     }, retryOptions);
   }
 
+  private static applyBlockHeight(request: RequestInit, url: string) {
+    const operationName = request.body
+      ?.toString()
+      .match(/"operationName":"(.+)"/)?.[1] as keyof SdkOperations;
+
+    if (!BLOCK_HEIGHT_SENSITIVE_OPERATIONS.includes(operationName)) {
+      return;
+    }
+
+    const normalizedUrl = url.replace(/-sub$/, '');
+    const currentBlockHeight = this.currentBlockHeightCache[normalizedUrl] ?? 0;
+
+    request.body = request.body
+      ?.toString()
+      .replace(/}$/, `,"extensions":{"required_fuel_block_height":${currentBlockHeight}}}`);
+  }
+
+  private static async fetchAndProcessBlockHeight(
+    url: string,
+    request: RequestInit,
+    options: ProviderOptions
+  ): Promise<Response> {
+    const fetchFn = () =>
+      options.fetch ? options.fetch(url, request, options) : fetch(url, request);
+
+    let response: Response = await fetchFn();
+
+    const retryOptions: RetryOptions = {
+      maxRetries: 5,
+      baseDelay: 500,
+    };
```

Comment:
So am I correct in assuming these are fixed default values that the user cannot modify?

Reply:
Yes, you're right.